src/app/modules/addresses/addresses.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { Address } from 'src/app/models/address.model';
+import { selectCurrentCustomerOrganization } from 'src/app/store/current-customer/current-customer.selectors';
+import { getAddressess } from './store/addresses.actions';
+import { selectOrganizationAddresses } from './store/addresses.selectors';
+import { faChevronLeft, faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { pageSize } from './addresses.constants';
+
+@Component({
+  selector: 'vc-addresses',
+  templateUrl: './addresses.component.html',
+  styleUrls: [
+    './addresses.component.scss',
+  ],
+})
+export class AddressesComponent implements OnInit, OnDestroy {
+  unsubscriber = new Subject();
+
+  curentCustomerOrganization$ = this.store.select(selectCurrentCustomerOrganization);
+
+  curentCustomerOrganizationId?: string;
+
+  addresses?: Address[] | null;
+
+  totalCount?: number | null;
+
+  cursor = '0';
+
+  page = 1;
+
+  pageSize = pageSize;
+
+  faChevronLeft = faChevronLeft;
+
+  faChevronRight = faChevronRight;
+
+  sortDescending = 'desc';
+
+  sortAscending = 'asc';
+
+  sortDirection = this.sortDescending;
+
+  constructor(private readonly store: Store) { }
+
+  ngOnInit(): void {
+    this.curentCustomerOrganization$.subscribe(value => this.curentCustomerOrganizationId = value?.id);
```

Comment:
We assume current organization is always exists and we prevent someone go to this page by `if` on dashboard. I'll prefer use `nonNull` type guard here instead (in `pipe`) and remove `if` check below. This component will broken any way if someone will go here as anonymous user

Reply:
Also there is no ubsubscription

Reply:
done